### ISSUE TITLE
Macro timing out after seemingly successful timeout increase

### ISSUE THREAD

**Issue by ronpastore** [xlwings/xlwings#2297]
#### Describe your issue (incl. Traceback!)
I'm setting the timeout ( correctly, i think) and i'm still timing out around 1:25 seconds for long running macros. 
```python
app = xlwings.App(visible=True, add_book=False)
with app.properties(timeout=6000) as app2:
    book = app2.books.open(location)
    sheet = book.sheets("Sheet Name")

    print("timeout ->", app2.timeout)  # prints 6000 
    
    m = book.macro('Solver_Name')
    m()
```

```python
# Your traceback here
Command Failed:
    OSERROR: -1712
    MESSAGE: Apple event timed out.
    COMMAND: app('/Application/Microsoft Excel.app', newinstance=(0, 18448791)).run_VB_macro("'filename.xlsm'!Solver_Name")
```

**Comment by fzumstein** [comment]
You can only use app.properties on, well, properties of app (https://docs.xlwings.org/en/latest/api/app.html#xlwings.App). `timeout` isn't one of them.

Might be resolved by including `timeout=-1` here: https://github.com/xlwings/xlwings/blob/main/xlwings/_xlmac.py#L341
as it's done e.g., here: https://github.com/xlwings/xlwings/blob/main/xlwings/_xlmac.py#L527

**Comment by ronpastore** [comment]
> You can only use app.properties on, well, properties of app (https://docs.xlwings.org/en/latest/api/app.html#xlwings.App). `timeout` isn't one of them.
> 
> Might be resolved by including `timeout=-1` here: https://github.com/xlwings/xlwings/blob/main/xlwings/_xlmac.py#L341 as it's done e.g., here: https://github.com/xlwings/xlwings/blob/main/xlwings/_xlmac.py#L527

thanks, it doesn't look like there is a timeout accepted on Macro.run, also trying Book.macro but same, 'unexpected keyword'.

**Comment by fzumstein** [comment]
No, this has to be fixed in the linked source code, it's not an option in the public API.

**Comment by ronpastore** [comment]
> No, this has to be fixed in the linked source code, it's not an option in the public API.

Might not be public API but seems like calling like this might be working, just ran over 2 mins, 
```python
macro.app.impl.xl.run_VB_macro(<MACRO NAME>, timeout=-1)```

**Comment by fzumstein** [comment]
Good idea, but you'd want to use the official API for the workaround:
```python
macro.app.api.run_VB_macro(<MACRO NAME>, timeout=-1)
```

**Comment by ronpastore** [comment]
> Good idea, but you'd want to use the official API for the workaround:
> 
> ```python
> macro.app.api.run_VB_macro(<MACRO NAME>, timeout=-1)
> ```

oh great, didn't think this was possible, will give it a shot.
